src/nti/contentfragments/html.py

```diff
@@ -233,6 +235,13 @@ def sanitize_token(self, token):
             # character data beneath a rejected element
             return None
 
+        # Indicate whether we're in an anchor tag
+        if token.get('name') == 'a':
+            if token_type == 'StartTag':
+                self._in_anchor = True
+            elif token_type == 'EndTag':
+                self._in_anchor = False
```

Comment:
I was tempted to shorten this to just `self._in_anchor = token_type == 'StartTag'` and it took a bit to dawn on me why that's not right. Maybe a quick comment about why that's no good?

Reply:
Huh, well I'd assumed EndTag would cause issues, but I don't seem to be able to cause issues with it, as an empty anchor tag with url-like text following seems to get turned into a starttag-characters-endtag sequence, e.g. `<a href="http://www.google.com"/>www.google.com` effectively becomes `<a href="http://www.google.com">www.google.com</a>`.  Is there another reason to avoid that?

Reply:
If you ever get another token type (that has a `name`) besides `StartTag` or `EndTag`, then the shortcut syntax would be broken (I can't remember what the token types are or whether their data ever includes 'name'; I suspect most readers of the code won't remember either).